test/hotspot/jtreg/compiler/c2/irTests/TestVectorizationMismatchedAccess.java

```diff
@@ -153,54 +165,68 @@ static private void runAndVerify3(Runnable test, int offset) {
 
     @Test
     @IR(counts = { IRNode.LOAD_VECTOR_L, ">=1", IRNode.STORE_VECTOR, ">=1" },
-        // This test fails with compact headers, but only with UseSSE<=3.
-        applyIf = { "UseCompactObjectHeaders", "false" },
+        applyIfOr = {"UseCompactObjectHeaders", "false", "AlignVector", "false"},
         applyIfCPUFeatureOr = {"sse2", "true", "asimd", "true"},
         applyIfPlatform = {"64-bit", "true"})
     // 32-bit: offsets are badly aligned (UNSAFE.ARRAY_BYTE_BASE_OFFSET is 4 byte aligned, but not 8 byte aligned).
     //         might get fixed with JDK-8325155.
     public static void testByteLong1a(byte[] dest, long[] src) {
         for (int i = 0; i < src.length; i++) {
             UNSAFE.putLongUnaligned(dest, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 8 * i, handleByteOrder(src[i]));
+            // With AlignVector, we need 8-byte alignment of vector loads/stores.
+            // UseCompactObjectHeaders=false                  UseCompactObjectHeaders=true
+            // B_adr = base + 16 + 8*i  ->  always            B_adr = base + 12 + 8*i  ->  never
+            // L_adr = base + 16 + 8*i  ->  always            L_adr = base + 16 + 8*i  ->  always
+            // -> vectorize                                   -> no vectorization
         }
     }
 
     @Test
     @IR(counts = { IRNode.LOAD_VECTOR_L, ">=1", IRNode.STORE_VECTOR, ">=1" },
-        // This test fails with compact headers, but only with UseSSE<=3.
-        applyIf = { "UseCompactObjectHeaders", "false" },
+        applyIfOr = {"UseCompactObjectHeaders", "false", "AlignVector", "false"},
         applyIfCPUFeatureOr = {"sse2", "true", "asimd", "true"},
         applyIfPlatform = {"64-bit", "true"})
     // 32-bit: address has ConvL2I for cast of long to address, not supported.
     public static void testByteLong1b(byte[] dest, long[] src) {
         for (int i = 0; i < src.length; i++) {
             UNSAFE.putLongUnaligned(dest, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 8L * i, handleByteOrder(src[i]));
+            // With AlignVector, we need 8-byte alignment of vector loads/stores.
+            // UseCompactObjectHeaders=false                  UseCompactObjectHeaders=true
+            // B_adr = base + 16 + 8*i  ->  always            B_adr = base + 12 + 8*i  ->  never
+            // L_adr = base + 16 + 8*i  ->  always            L_adr = base + 16 + 8*i  ->  always
+            // -> vectorize                                   -> no vectorization
         }
     }
 
     @Test
     @IR(counts = { IRNode.LOAD_VECTOR_L, ">=1", IRNode.STORE_VECTOR, ">=1" },
-        // This test fails with compact headers, but only with UseSSE<=3.
-        applyIf = { "UseCompactObjectHeaders", "false" },
```

Comment:
Is this removed intentiionally? or it should be `applyIfOr = {"UseCompactObjectHeaders", "false", "AlignVector", "false"},`?

Reply:
Well, it turns out that this case always vectorizes. Did you see the comments below? It is a strange exception, because this test manually sets a large "base-to-payload-offset" of 64.